scanner/src/main/kotlin/Scanner.kt

```diff
@@ -50,7 +51,11 @@ const val TOOL_NAME = "scanner"
  * The class to run license / copyright scanners. The signatures of public functions in this class define the library
  * API.
  */
-abstract class Scanner(val scannerName: String, protected val scannerConfig: ScannerConfiguration) {
+abstract class Scanner(
+    val scannerName: String,
+    protected val scannerConfig: ScannerConfiguration,
+    protected val downloaderConfig: DownloaderConfiguration
+) {
```

Comment:
@fviernau @mnonnenmacher @sschuberth please have a look at e04d6d33b5d5d7a8cf0d7af173950ac1f48c884b.
As the Scanner implicitly calls the Downloader, the `DownloaderConfiguration` is either needed in the Scanner (see: e04d6d33b5d5d7a8cf0d7af173950ac1f48c884b), or we need a way to access the ORT config / downloader config from the `Downloader` directly and not only via `DownloaderCommand`.
I see the issue that this would be problematic when using the Scanner as a library function.

What do you think. Any suggestions how to implement that?

Reply:
> I see the issue that this would be problematic when using the Scanner as a library function.

Why? I think as the scanner depends on the downloader it's ok that it requires a downloader config.

Reply:
But the fact that the scanner depends on the downloader is (somewhat) an implementation detail. I wonder whether it would be nicer to pass the *whole* ORT config to the scanner (and each other module), and then let each module pick only the parts it needs.